crates/accelerate/src/basis/basis_translator/mod.rs

```diff
@@ -10,14 +10,804 @@
+                    node_obj.extra_attrs.clone(),
+                    #[cfg(feature = "cache_pygates")]
+                    None,
+                )?;
+            }
+            continue;
+        }
+        let node_qarg_as_physical: Option<Qargs> =
+            Some(node_qarg.iter().map(|x| PhysicalQubit(x.0)).collect());
+        if qargs_with_non_global_operation.contains_key(&node_qarg_as_physical)
+            && qargs_with_non_global_operation[&node_qarg_as_physical].contains(node_obj.op.name())
+        {
+            // out_dag.push_back(py, node_obj)?;
+            out_dag.apply_operation_back(
+                py,
+                node_obj.op.clone(),
+                node_qarg,
+                node_carg,
+                if node_obj.params_view().is_empty() {
+                    None
+                } else {
+                    Some(
+                        node_obj
+                            .params_view()
+                            .iter()
+                            .map(|param| param.clone_ref(py))
+                            .collect(),
+                    )
+                },
+                node_obj.extra_attrs.clone(),
+                #[cfg(feature = "cache_pygates")]
+                None,
+            )?;
+            continue;
+        }
+
+        if dag.has_calibration_for_index(py, node)? {
+            out_dag.apply_operation_back(
+                py,
+                node_obj.op.clone(),
+                node_qarg,
+                node_carg,
+                if node_obj.params_view().is_empty() {
+                    None
+                } else {
+                    Some(
+                        node_obj
+                            .params_view()
+                            .iter()
+                            .map(|param| param.clone_ref(py))
+                            .collect(),
+                    )
+                },
+                node_obj.extra_attrs.clone(),
+                #[cfg(feature = "cache_pygates")]
+                None,
+            )?;
+            continue;
+        }
+        let unique_qargs: Option<Qargs> = if qubit_set.is_empty() {
+            None
+        } else {
+            Some(qubit_set.iter().map(|x| PhysicalQubit(x.0)).collect())
+        };
+        if extra_inst_map.contains_key(&unique_qargs) {
+            replace_node(
+                py,
+                &mut out_dag,
+                node_obj.clone(),
+                &extra_inst_map[&unique_qargs],
+            )?;
+        } else if instr_map
+            .contains_key(&(node_obj.op.name().to_string(), node_obj.op.num_qubits()))
+        {
+            replace_node(py, &mut out_dag, node_obj.clone(), instr_map)?;
+        } else {
+            return Err(TranspilerError::new_err(format!(
+                "BasisTranslator did not map {}",
+                node_obj.op.name()
+            )));
+        }
+        is_updated = true;
+    }
+
+    Ok((out_dag, is_updated))
+}
+
+fn replace_node(
```

Comment:
I'm not quite sure, especially because of the instr_map argument (Which could easily be changed to work with another `DAGCircuit` and `Param` collection). One thing I'm certain of is that if we do add it, we need to change the name as it is a little misleading because we're not replacing any nodes but rather copying over nodes.